src/react-components/object-info-dialog.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import DialogContainer from "./dialog-container.js";
+import styles from "../assets/stylesheets/client-info-dialog.scss";
+import { FormattedMessage } from "react-intl";
+
+export default class ObjectInfoDialog extends Component {
+  static propTypes = {
+    scene: PropTypes.object,
+    el: PropTypes.object,
+    objectDisplayString: PropTypes.string,
+    onClose: PropTypes.func,
+    hubChannel: PropTypes.object
+  };
+
+  state = {
+    pinned: false
+  };
+
+  componentDidMount() {
+    this.updatePinnedState = this.updatePinnedState.bind(this);
+    this.pin = this.pin.bind(this);
+    this.unpin = this.unpin.bind(this);
+    this.props.scene.addEventListener("uninspect", this.props.onClose);
+    this.updatePinnedState();
+  }
+
+  updatePinnedState() {
+    this.setState({ pinned: this.props.el.components["networked"].data.persistent });
+  }
+
+  pin() {
+    if (!NAF.utils.isMine(this.props.el) && !NAF.utils.takeOwnership(this.props.el)) return;
+    this.props.el.emit("pinned", { el: this.props.el });
+    this.props.el.setAttribute("pinnable", "pinned", true);
+    this.updatePinnedState();
+  }
+
+  unpin() {
+    if (!NAF.utils.isMine(this.props.el) && !NAF.utils.takeOwnership(this.props.el)) return;
+    this.props.el.emit("unpinned", { el: this.props.el });
+    this.props.el.setAttribute("pinnable", "pinned", false);
+    this.updatePinnedState();
+  }
+
+  delete() {
+    const targetEl = this.props.el;
+
+    if (!NAF.utils.isMine(targetEl) && !NAF.utils.takeOwnership(targetEl)) return;
+
+    targetEl.setAttribute("animation__remove", {
+      property: "scale",
+      dur: 200,
+      to: { x: 0.01, y: 0.01, z: 0.01 },
+      easing: "easeInQuad"
+    });
+
+    targetEl.addEventListener("animationcomplete", () => {
+      this.props.scene.systems["hubs-systems"].cameraSystem.uninspect();
+      NAF.utils.takeOwnership(targetEl);
```

Comment:
It feels really odd to me that we would take ownership _after_ the animation. Ideally I think we would immediately take ownership and delete the object, but still somehow play the animation locally. Unfortunately I don't think we have an easy way to delete objects over the network but _not_ delete them locally.

Reply:
Here we take ownership before AND after the animation, copying this from `remove-networked-object-button`. (If taking ownership fails above, then we early out.) Perhaps though I should be checking that the `targetEl` still exists and I still have ownership (or can take ownership back) in the case that I leave pause mode during the animation and a delete message came across the network while I was paused.